devito/passes/iet/mpi.py

```diff
@@ -79,7 +79,8 @@ def rule1(dep, candidates, loc_dims):
     rules = [rule0, rule1]
 
     # Precompute scopes to save time
-    scopes = {i: Scope([e.expr for e in v]) for i, v in MapNodes().visit(iet).items()}
+    scopes = {i: Scope([e.expr for e in v if not isinstance(e, Call)])
+              for i, v in MapNodes().visit(iet).items()}
 
```

Comment:
This is because there are now `Call` objects appearing inside the time loop to run the solver